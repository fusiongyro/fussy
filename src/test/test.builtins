/* -*-C++-*- */
/******************************************************************
 * Copyright (c) 2000-2018, 2019 S.Bhatnagar
 *
 * This program is free software; you can redistribute it and/or modify
 * it under the terms of the GNU General Public License as published by
 * the Free Software Foundation; either version 2 of the License, or
 * (at your option) any later version.
 *
 * This program is distributed in the hope that it will be useful,
 * but WITHOUT ANY WARRANTY; without even the implied warranty of
 * MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the
 * GNU General Public License for more details.
 *
 * You should have received a copy of the GNU General Public License
 * along with this program; if not, write to the Free Software
 * Foundation, Inc., 59 Temple Place, Suite 330, Boston, MA  02111-1307  USA
 *

  History:

    First version.

               Sanjay Bhatnagar, Dec 2001
               sanjay@ncra.tifr.res.in

******************************************************************/
test(){
auto x,y,x1,x2
x=1pm0.1
y=2pm0.2
x1=1pm0.1
x2=1pm0.1
print "Value of x=\t",x,"\n"
print "Value of y=\t",y,"\n"
print "Value of x1=\t",x1,"\n"
print "Value of x2=\t",x2,"\n"
print "sin(x)=\t",sin(x),"\n"
print "sqrt(1-sin(x)^2)=\t",sqrt(1-sin(x)^2),"\n"
print "cos(x)=\t",cos(x),"\n"
print "tan(x)=\t",tan(x),"\n"
print "sin(x)/cos(x)=\t",sin(x)/cos(x),"\n"

print "asin(sin(x))=\t",asin(sin(x)),"\n"
print "acos(cos(x))=\t",acos(cos(x)),"\n"
print "atan(tan(x))=\t",atan(tan(x)),"\n"

print "asinh(sinh(x))=\t",asinh(sinh(x)),"\n"
print "acosh(cosh(x))=\t",acosh(cosh(x)),"\n"
print "atanh(tanh(x))=\t",atanh(tanh(x)),"\n"

print "exp(ln(x))=\t",exp(ln(x)),"\n";
print "sinh(x)=\t",sinh(x),"\n"
print "(exp(x)-exp(-x))/2=\t",(exp(x)-exp(-x))/2,"\n"
print "sinh(x)/((exp(x)-exp(-x))/2)=\t",sinh(x)/((exp(x)-exp(-x))/2),"\n"
print "acosh(cosh(x))=\t",acosh(cosh(x)),"\n"
print "atanh(tanh(x))=\t",atanh(tanh(x)),"\n"

print "exp(ln(x))=",exp(ln(x)),"\n"
print "x/exp(ln(x))=",x/exp(ln(x)),"\n"

print "y/(1-y) =",(y/(1-y))%16.6f,"\n";
print "y/(-y+1)=",(y/(-y+1))%16.6f,"\n"

print "sin(x1)*sin(x1)=\t",sin(x1)*sin(x1),"\n";
print "sin(x1)*sin(x2)=\t",sin(x1)*sin(x2),"\n";
print "(sin(x1)^2+cos(x1)^2)=\t",(sin(x1)^2+cos(x1)^2),"\n";
print "(sin(x1)^2+cos(x1)^2)/(sin(x2)^2+cos(x2)^2)=\t",(sin(x1)^2+cos(x1)^2)/(sin(x2)^2+cos(x2)^2),"\n";
print "sin(x1)^2+cos(x2)^2=\t",sin(x1)^2+cos(x2)^2,"\n";
};
test();
