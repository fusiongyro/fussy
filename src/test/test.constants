/* -*-C++-*- */
/******************************************************************
 * Copyright (c) 2000-2017, 2018 S.Bhatnagar
 *
 * This program is free software; you can redistribute it and/or modify
 * it under the terms of the GNU General Public License as published by
 * the Free Software Foundation; either version 2 of the License, or
 * (at your option) any later version.
 *
 * This program is distributed in the hope that it will be useful,
 * but WITHOUT ANY WARRANTY; without even the implied warranty of
 * MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the
 * GNU General Public License for more details.
 *
 * You should have received a copy of the GNU General Public License
 * along with this program; if not, write to the Free Software
 * Foundation, Inc., 59 Temple Place, Suite 330, Boston, MA  02111-1307  USA
 *

  History:

    First version.

               Sanjay Bhatnagar, Dec 2001
               sanjay@ncra.tifr.res.in

******************************************************************/
test()
{
  auto x,y;
  print "10pm1+10pm1 = ",10pm1+10pm1,"\n";
  print "2*10pm1    = ",2*10pm1,"\n";
  x=10pm1;
  y:=x;
  print "x=",x," y:=x = ",y,"\n";
  print "x+x=",x+x,"\n";
  print "x+y=",x+y,"\n";;
}
f(){return 10pm1;}
g(){return 10pm1;}
test();
print "f()*g() = ",f()*g(),"\n";
