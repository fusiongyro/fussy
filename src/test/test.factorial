/* -*-C++-*- */
/******************************************************************
 * Copyright (c) 2000-2017, 2018 S.Bhatnagar
 *
 * This program is free software; you can redistribute it and/or modify
 * it under the terms of the GNU General Public License as published by
 * the Free Software Foundation; either version 2 of the License, or
 * (at your option) any later version.
 *
 * This program is distributed in the hope that it will be useful,
 * but WITHOUT ANY WARRANTY; without even the implied warranty of
 * MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the
 * GNU General Public License for more details.
 *
 * You should have received a copy of the GNU General Public License
 * along with this program; if not, write to the Free Software
 * Foundation, Inc., 59 Temple Place, Suite 330, Boston, MA  02111-1307  USA
 *

  History:

    First version.

               Sanjay Bhatnagar, Dec 2001
               sanjay@ncra.tifr.res.in

******************************************************************/
/*
  df(x)
  ----   = f(x) [1/x + 1/(x-1) + 1/(x-2) +...+ 1]
  dx
*/
f(x)
{
  if (x==1) {return x;} else {return x*f(--x);}
}

df(x)
{
  if (x==1) {return x;} else {return 1/x+df(--x);};
}

test()
{
  auto x,i;
  for(i=1;i<20;i++) {x=i+1pm1;print x," ",f(x),"\n";};
}

test();
