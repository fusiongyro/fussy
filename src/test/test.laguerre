/* -*-C++-*- */
/******************************************************************
 * Copyright (c) 2000-2018, 2019 S.Bhatnagar
 *
 * This program is free software; you can redistribute it and/or modify
 * it under the terms of the GNU General Public License as published by
 * the Free Software Foundation; either version 2 of the License, or
 * (at your option) any later version.
 *
 * This program is distributed in the hope that it will be useful,
 * but WITHOUT ANY WARRANTY; without even the implied warranty of
 * MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the
 * GNU General Public License for more details.
 *
 * You should have received a copy of the GNU General Public License
 * along with this program; if not, write to the Free Software
 * Foundation, Inc., 59 Temple Place, Suite 330, Boston, MA  02111-1307  USA
 *

  History:

    First version.

               Sanjay Bhatnagar, Dec 2001
               sanjay@ncra.tifr.res.in

******************************************************************/
/*
    Recursive generating formula for Laguerre polynomial and it's
    derivative.

    Shows that f(n,x).rms == (df(n,x).val*x.rms) (which is
    f'(n,x)*deltaX) 
*/

f(n,x)
{
  if (n<=0) return 1; 
  if (n==1) return -x+1; 
  return ((2*n-1-x)*f(n-1,x)-(n-1)*f(n-2,x))/n;
}

df(n,x)
{
  return (n*f(n,x)-n*f(n-1,x))/x;
}

test(x)
{
   auto n,i,y;
   for(i=1;i<10;i++) {print "l(",i%-6.2f,",",x%-6.2f,")= ",f(i,x)," ",fabs(df(i,x).val*x.rms),"\n";};
}

test(3pm1);
