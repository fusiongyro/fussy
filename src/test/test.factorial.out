/****************************************************************************/
/* Copyright (c) 2000-2017, 2018 S.Bhatnagar                                */
/*                                                                          */
/* This program is free software; you can redistribute it and/or modify     */
/* it under the terms of the GNU General Public License as published by     */
/* the Free Software Foundation; either version 2 of the License, or        */
/* (at your option) any later version.                                      */
/*                                                                          */
/* This program is distributed in the hope that it will be useful,          */
/* but WITHOUT ANY WARRANTY; without even the implied warranty of           */
/* MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the            */
/* GNU General Public License for more details.                             */
/*                                                                          */
/* You should have received a copy of the GNU General Public License        */
/* along with this program; if not, write to the Free Software              */
/* Foundation, Inc., 59 Temple Place, Suite 330, Boston, MA  02111-1307 USA */
/****************************************************************************/
   2.00000 +/-    1.00000    2.00000 +/-    3.00000
   3.00000 +/-    1.00000    6.00000 +/-   11.00000
   4.00000 +/-    1.00000   24.00000 +/-   50.00000
   5.00000 +/-    1.00000  120.00000 +/-  274.00000
   6.00000 +/-    1.00000  720.00000 +/- 1764.00000
   7.00000 +/-    1.00000 5040.00000 +/- 13068.00000
   8.00000 +/-    1.00000 40320.00000 +/- 109584.00000
   9.00000 +/-    1.00000 362880.00000 +/- 1026576.00000
  10.00000 +/-    1.00000 3628800.00000 +/- 10628640.00000
  11.00000 +/-    1.00000 39916800.00000 +/- 120543840.00000
  12.00000 +/-    1.00000 479001600.00000 +/- 1486442880.00000
  13.00000 +/-    1.00000 6227020800.00000 +/- 19802759040.00000
  14.00000 +/-    1.00000 87178291200.00000 +/- 283465647360.00000
  15.00000 +/-    1.00000 1307674368000.00000 +/- 4339163001600.00000
  16.00000 +/-    1.00000 20922789888000.00000 +/- 70734282393600.00000
  17.00000 +/-    1.00000 355687428096000.00000 +/- 1223405590579200.00000
  18.00000 +/-    1.00000 6402373705728000.00000 +/- 22376988058521600.00000
  19.00000 +/-    1.00000 121645100408832000.00000 +/- 431565146817638400.00000
  20.00000 +/-    1.00000 2432902008176640000.00000 +/- 8752948036761600000.00000
